src/sonic-yang-models/yang-models/sonic-vxlan.yang

```diff
@@ -0,0 +1,205 @@
+module sonic-vxlan {
+    namespace "http://github.com/Azure/sonic-vxlan";
+    prefix svxlan;
+    yang-version 1.1;
+
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+
+    import sonic-extension {
+        prefix sonic-ext;
+    }
+    import sonic-vlan {
+        prefix svlan;
+    }
+
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONIC VXLAN";
+
+    revision 2021-04-12 {
+        description
+            "First revision.";
+    }
+
+    container sonic-vxlan {
+
+        container VXLAN_TUNNEL {
+
+            list VXLAN_TUNNEL_LIST {
+                key "name";
+                max-elements 1;
+
+                leaf name {
+                    must "(starts-with(../name, 'vtep'))"
+                    {
+                        error-message "Error: Name should start with vtep";
+                        error-app-tag invalid-vtep-name;
+                    }
+                    type string {
+                        length 1..10;
```

Comment:
>1..10 [](http://example.com/codeflow?start=31&length=5)

Where is the limitation from?

Reply:
for every vlan vni map a vni netdevice of form <vtepName>-<vlanid> is created. 
The kernel devices can have a max length of 15 characters.
Keeping aside 5 characters for the "-<vlanid>" we are left with 10.

Reply:
Using "-" is a must or the user can choose any letter after "vtep" prefix? if any character, plese put max 15 chars.

Reply:
The vni device format is <tunnelname>-<vid>
Max vid is 4 characters and 1 for hyphen leaving us 
with 10 characters for tunnelname.

Added a comment in the yang file for the same.